File: 2021/ts/two/instruction.ts
export enum Direction {
  Down = "down",
  Forward = "forward",
  Up = "up",
}

interface Instruction {
  direction: Direction;
  value: number;
}

export function getInstruction(input: string): Instruction {
  const [rawDirection, rawValue] = input.split(" ");
  return {
    direction: rawDirection as Direction,
    value: parseInt(rawValue, 10),
  };
}

File: 2021/rust/two/src/instruction.rs
pub mod instruction {
    pub enum Direction {
        Down,
        Forward,
        Up,
    }

    pub struct Instruction {
        pub direction: Direction,
        pub value: i32,
    }

    impl Instruction {
        pub fn new(input: &str) -> Instruction {
            let instruction = input.split_whitespace().collect::<Vec<_>>();
            Instruction {
                direction: match instruction[0] {
                    "forward" => Direction::Forward,
                    "down" => Direction::Down,
                    "up" => Direction::Up,
                    _ => panic!("Unknown direction."),
                },
                value: instruction[1].parse::<i32>().unwrap(),
            }
        }
    }
}
